src/main/minio.js

```diff
@@ -244,20 +248,26 @@ export class Client {
     this.userAgent = `${this.userAgent} ${appName}/${appVersion}`
   }
 
-  // partSize will be atleast minimumPartSize or a multiple of minimumPartSize
-  // for size <= 50GiB partSize is always 5MiB (10000*5MiB = 50GiB)
-  // for size > 50GiB partSize will be a multiple of 5MiB
-  // for size = 5TiB partSize will be 525MiB
+  // Calculate part size given the object size. Part size will be atleast this.partSize
   calculatePartSize(size) {
     if (!isNumber(size)) {
       throw new TypeError('size should be of type "number"')
     }
     if (size > this.maxObjectSize) {
       throw new TypeError(`size should not be more than ${this.maxObjectSize}`)
     }
-    var partSize = Math.ceil(size/10000)
-    partSize = Math.ceil(partSize/this.minimumPartSize) * this.minimumPartSize
-    return partSize
+    if (this.overRidePartSize) {
+      return this.partSize
+    }
+    var partSize = this.partSize
+    while (true) {
+      // If partSize is big enough to accomodate the object size, then use it.
+      if ((partSize * 10000) > size) {
+	return partSize
+      }
+      // Try part sizes as 64MB, 90MB, 106MB etc.
```

Comment:
`// Try part sizes as 64MB, 90MB, 106MB etc.` should be `// Try part sizes as 64MB, 80MB, 96MB etc.`